vmdb/app/models/ems_amazon.rb

```diff
@@ -49,13 +49,18 @@ def verify_credentials(auth_type=nil, options={})
     begin
       # EC2 does Lazy Connections, so call a cheap function
       with_provider_connection(options.merge(:auth_type => auth_type)) { |ec2| ec2.regions.map(&:name) }
-    rescue AWS::EC2::Errors::SignatureDoesNotMatch
+    rescue AWS::EC2::Errors::SignatureDoesNotMatch => err
+      $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
       raise MiqException::MiqHostError, "SignatureMismatch - check your AWS Secret Access Key and signing method"
-    rescue AWS::EC2::Errors::AuthFailure
+    rescue AWS::EC2::Errors::AuthFailure => err
+      $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
       raise MiqException::MiqHostError, "Login failed due to a bad username or password."
+    rescue AWS::Errors::MissingCredentialsError => err
+      $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
+      raise MiqException::MiqHostError, "Missing credentials"
     rescue Exception => err
       $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
-      raise MiqException::MiqHostError, "Unexpected response returned from system, see log for details"
+      raise MiqException::MiqHostError, "Unexpected response returned from system: #{err.message}"
     end
```

Comment:
@himdel Again, this is much later on... but as reference, the whole concept of raising an exception to the UI seems weird now.

In #1990, I want to return an object from the former `verify_credentials` type calls that the UI can ask if it's `good?` or `bad?` and provide a `message` method so the UI can do things like check for `bad?` and present the error by calling `message` instead of having exception handling in both the UI and backend code.